msprime/mutations.py

```
@@ -199,6 +199,301 @@ def __init__(self, state_independent=False):
+                    beta_microsat(i, s, lo)
+                    * alpha_microsat(i, u, v, lo)
+                    * (1 - p)
+                    * gamma_microsat(m, i, j, lo, hi)
+                )
+            elif i == j + 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * (1 - alpha_microsat(i, u, v, lo))
+                    * (p + ((1 - p) * gamma_microsat(m, i, j, lo, hi)))
+                )
+            elif i > j + 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * (1 - alpha_microsat(i, u, v, lo))
+                    * (1 - p)
+                    * gamma_microsat(m, i, j, lo, hi)
+                )
+            else:
+                rate_matrix[ii, jj] = 0
+    # scale by max row sum; then normalize to 1
+    row_sums = rate_matrix.sum(axis=1, dtype="float64")
+    alpha = max(row_sums)
+    rate_matrix /= alpha
+    row_sums = rate_matrix.sum(axis=1, dtype="float64")
+    np.fill_diagonal(rate_matrix, 1.0 - row_sums)
+    return rate_matrix
+
+
+class MicrosatMutationModel(MatrixMutationModel):
+    """
+    General class for Microsatellite mutation models
+    which parameterizes the transition matrix describing
+    changes among alleles using 5 parameters according
+    to the model of Sainudiin et al. (2004).
+
+    Concretely, mutation rates under this model are computed as follows.
+    Let ``qij`` be the transition rate from equation (3) in Sainudiin et al,
+    and let ``M`` be the largest row sum of the matrix whose i,j-th entry
+    is ``qij`` and whose diagonals are zero. Then if the mutation rate
+    passed to :func:`.sim_mutations` is ``m``, the mutation rate from
+    allele ``i`` to allele `j`` is ``m * qij / M``.
+
+    :param float s: strength of length dependence on mutation rate
+        (i.e. s=0 uniform rate)
+    :param float u: constant bias parameter; must be between 0 and 1
```

We should probably raise an error if u is not between 0 and 1?